File: Leetcode9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0){
            return false;
        }
        if(x < 10){
            return true;
        }
        int num = x;
        long long reversed = 0;
        while(num != 0){
            int tempDig = num%10;
            num -= tempDig;
            num /= 10;
            reversed = (reversed * 10) + tempDig;
        }
        return reversed == x;
    }
};

File: Leetcode9.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0:
            return False
        numStr = str(x)
        for i in range(len(numStr) // 2):
            if numStr[i] != numStr[-i - 1]:
                return False
        return True
